src/wallet/wallet.cpp

```diff
@@ -1344,6 +1344,21 @@ void CWalletTx::GetAmounts(list<COutputEntry>& listReceived,
         nFee = nDebit - nValueOut;
     }
 
+    // Create an output for the value taken from the transparent value pool by JoinSplits
+    if (nDebit > 0) {
+        CAmount nValueOut = 0;
+        for (const JSDescription & js : vjoinsplit) {
+            nValueOut += js.vpub_old;
```

Comment:
Going back to the drawing board on this one since not all joinsplits may belong to our wallet which complicates things a fair bit.